tests/framework/extensions/hardening/k3s/harden_nodes.go

```diff
@@ -0,0 +1,81 @@
+package hardening
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+func CheckClusterHardening(client *rancher.Client, hardened bool, nodes []*nodes.Node, nodeRoles []string) error {
+	if hardened {
```

Comment:
If this is checking a hardened cluster, why do need this bool check?

Reply:
Would the checks be different from RKE2? Just curious, I do like the design for this.

Reply:
I think Anu is checking if it needs to be different for RKE2, as of now we don't believe it needs to be? I originally had that check to see if the `hardened` value in the user's config.yaml is true. If so, then it would proceed with the hardening , but I can definitely rework this if it seems a bit weird